content/articles/2018-07-ggplot2-3-0-0.Rmarkdown

```diff
@@ -0,0 +1,173 @@
+  annotate("point", x = -80, y = 35, colour = "red", size = 4)
+```
+
+### Calculated aesthetics
+
+The new `stat()` function offers a cleaner, and better-documented syntax for 
+calculated-aesthetic variables. This replaces the older approach of surrounding 
+the variable name with `...`. Instead of using `aes(y = ..count..)`, you can use 
+`aes(y = stat(count))`. 
+
+This is particularly nice for more complex calculations, as `stat()` only needs 
+to be specified once; e.g. `aes(y = stat(count / max(count)))` rather than 
+`aes(y = ..count.. / max(..count..))`.
+
+### Tag
+
+In addition to title, subtitle, and caption, a new tag label has been added, for identifying plots. Add a tag with `labs(tag = "A")`, style it with the `plot.tag` theme element, and control position with the `plot.tag.position` theme.
+
+```{r tag}
+ggplot(mtcars) +
+  geom_point(aes(disp, mpg)) + 
+  labs(tag = 'A', title = 'Title of this plot')
+```
+
+### Layers: geoms, stats, and position adjustments
+
+The new `position_dodge2()` allows you to arrange the horizontal position of plots with variable widths (i.e. bars and rectangles, in addition to box plots). By default, `position_dodge2()` preserves the width of each element.
+
+```{r dodge2-single}
+ggplot(mtcars, aes(factor(cyl), fill = factor(vs))) +
+  geom_bar(position = position_dodge2(preserve = "single"))
+```
+
+You can choose to preserve the total width by setting the `preserve` argument to `"total"`. Thank you to Kara Woo for all of her work on this.
+
+```{r dodge2-total}
+ggplot(mtcars, aes(factor(cyl), fill = factor(vs))) +
+  geom_bar(position = position_dodge2(preserve = "total"))
+```
+
+### Scales and guides
+
+The new `scale_alpha()` function can be used to but can be used to visually down-weight less important observations.
+
+```{r scale-alpha}
+ggplot(mpg, aes(displ, hwy)) +
+  geom_point(aes(alpha = year))
+```
+
+Several new functions have been added to make it easy to use Viridis colour 
+scales: `scale_colour_viridis_c()` and `scale_fill_viridis_c()` for continuous, 
+and `scale_colour_viridis_d()` and `scale_fill_viridis_d()` for discrete. 
+Viridis is also now used as the default colour and fill scale for ordered 
+factors.
+
+```{r viridis}
+dsamp <- diamonds[sample(nrow(diamonds), 1000), ]
+d <- ggplot(dsamp, aes(carat, price)) +
+  geom_point(aes(colour = clarity))
+d + scale_colour_viridis_d()
+```
+```{r viridis-c}
+v <- ggplot(faithfuld) +
+  geom_tile(aes(waiting, eruptions, fill = density))
+v + scale_fill_viridis_c()
+```
+
+
+`guide_colorbar()` is more configurable: tick marks and color bar frame can now by styled with arguments `ticks.colour`, `ticks.linewidth`, `frame.colour`, `frame.linewidth`, and `frame.linetype`.
+
+```{r guide-colorbar}
+set.seed(123)
+df <- data.frame(x <- rnorm(10),
+                 y <- rnorm(10),
+                 fill <- rnorm(10))
+
+p <- ggplot(df, aes(x, y, fill = fill)) + geom_point()
+p + scale_fill_gradient(low = 'white', high = 'red',
```

Comment:
Use tidyverse style guide 🙀

Reply:
Damn, that's what I get for copying and pasting!